apps/social/src/hooks/api.ts

```diff
@@ -0,0 +1,194 @@
+
+  return state?.Channel;
+}
+
+function useChannels() {
+  const payload = useMemo(() => ({ AllChannels: null }), []);
+  const { state, isStateRead } = useRouterState<ChannelsState>(payload);
+
+  return { channels: state?.AllChannels, isStateRead };
+}
+
+function useSubscriptions() {
+  const { account } = useAccount();
+  const actorId = account?.decodedAddress;
+  const payload = useMemo(() => ({ SubscribedToChannels: actorId }), [actorId]);
+
+  const { state, isStateRead } = useRouterState<SubscriptionState>(payload);
+
+  return { subscriptions: state?.SubscribedToChannels, readSubscriptions: isStateRead };
+}
+
+function useMessages() {
+  const [messages, setMessages] = useState<Message[] | null>();
+  const apiRequest = new ServerRPCRequestService();
+
+  const { id } = useParams() as Params;
+  const { api } = useApi();
+  const genesis = localStorage.getItem(LOCAL_STORAGE.GENESIS);
+
+  useEffect(() => {
+    apiRequest
+      .getResource('program.meta.get', [{ programId: id, genesis }])
+      .then(
+        ([
+          {
+            result: { metaFile },
+          },
+        ]) => Buffer.from(metaFile, 'base64'),
+      )
+      .then((buffer) => api.programState.read(id, buffer))
+      .then((state) => setMessages(state.toHuman() as Message[]));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  return { messages };
+}
+
+function useFeed() {
+  const [messages, setMessages] = useState<Message[] | null>();
+  const { channels } = useChannels();
+  const { api } = useApi();
+
+  useEffect(() => {
+    const getMessges = async () => {
+      const apiRequest = new ServerRPCRequestService();
+
+      const batchParams = channels!.map(({ id }) => ({
+        programId: id,
+        genesis: localStorage.getItem(LOCAL_STORAGE.GENESIS),
+      }));
+
+      const response = await apiRequest.getResource('program.meta.get', batchParams);
+
+      const promises = response
+        .filter((res: any) => res.result)
+        .map(async ({ result: { program, metaFile } }: any) => {
+          const buffer = Buffer.from(metaFile, 'base64');
+          const state = await api.programState.read(program, buffer);
+
+          return state.toHuman();
+        });
+
+      const messagaArr = await Promise.all(promises);
+      const msg = messagaArr.flat() as Message[];
+      const sorted = msg.sort(
+        (prev, next) =>
+          parseInt(prev.timestamp.replaceAll(',', ''), 10) - parseInt(next.timestamp.replaceAll(',', ''), 10),
+      );
+      setMessages(sorted.reverse());
+    };
+
+    if (channels) getMessges();
+  }, [channels, api.programState]);
+
+  return messages;
+}
+
+function useOwnFeed() {
+  const [messages, setMessages] = useState<Message[] | null>();
+  const { subscriptions } = useSubscriptions();
+  const { api } = useApi();
+
+  useEffect(() => {
+    const getMessges = async () => {
+      const apiRequest = new ServerRPCRequestService();
+
```

Comment:
looks like you can make separate hook, `useMessages` or something, since `useEffect` is doing the same thing as `useEffect` in `useFeed`